ddtrace/internal/_encoding.pyx

```diff
@@ -372,7 +373,7 @@ cdef class MsgpackEncoderBase(BufferedEncoder):
         self.max_size = max_size
         self.pk.buf_size = buf_size
         self.max_item_size = max_item_size if max_item_size < max_size else max_size
-        self._lock = threading.Lock()
+        self._lock = threading.RLock()
```

Comment:
Do we also need a lock around `ListBufferedEncoder.put`? (or well, move it?)

`encode_item` should probably also have a lock around it, but better to be sure it is around `put`?